tensorflow/python/ops/random_ops.py

```diff
@@ -310,21 +310,33 @@ def random_crop(value, size, seed=None, name=None):
   # If size > value.shape, zero pad the result so that it always has shape
   # exactly size.
   with ops.name_scope(name, "random_crop", [value, size]) as name:
+    size_dim = size
     value = ops.convert_to_tensor(value, name="value")
     size = ops.convert_to_tensor(size, dtype=dtypes.int32, name="size")
     shape = array_ops.shape(value)
-    check = control_flow_ops.Assert(
-        math_ops.reduce_all(shape >= size),
-        ["Need value.shape >= size, got ", shape, size],
-        summarize=1000)
-    shape = control_flow_ops.with_dependencies([check], shape)
-    limit = shape - size + 1
-    offset = random_uniform(
-        array_ops.shape(shape),
-        dtype=size.dtype,
-        maxval=size.dtype.max,
-        seed=seed) % limit
-    return array_ops.slice(value, offset, size, name=name)
+    if value.shape[0]:
+      if value.shape[0].compat.dimension_value() < size_dim[0]:
+        tmp = []
+        for ch in range(value.shape[2].value):
```

Comment:
Why do you have to pad each channel separately?

Reply:
Thanks, @alextp, what would you suggest to do it once?